amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/RetrievalSpecificConfig.java

```
@@ -15,11 +15,27 @@
 
 package software.amazon.kinesis.retrieval;
 
+import java.util.function.Function;
+import org.apache.commons.lang3.tuple.Pair;
+import software.amazon.kinesis.common.StreamIdentifier;
+import software.amazon.kinesis.retrieval.polling.DataFetcher;
+import software.amazon.kinesis.retrieval.polling.KinesisDataFetcher;
+
 public interface RetrievalSpecificConfig {
     /**
      * Creates and returns a retrieval factory for the specific configuration
-     * 
+     *
      * @return a retrieval factory that can create an appropriate retriever
      */
     RetrievalFactory retrievalFactory();
+
+
+    /**
+     * Creates and returns a retieval factory for the specific configuration based on provided data fetcher
+     *
+     * @return a retrieval factory that can create an appropriate retriever with a given data fetcher
+     */
+    default RetrievalFactory retrievalFactory(Function<Pair<KinesisDataFetcher, StreamIdentifier>, DataFetcher> dataFetcherProvider) {
+        return null;
```

Let's throw UnsupportedOperationException here